### PR TITLE
Fix #6204 - Coil:Cooling:DX:VariableSpeed air-to-air coil should autosize cooling capacity using CapFT curve results based on OutTemp and not RatedInletWaterTemp

### LINKED ISSUE #6204: Coil:Cooling:DX:VariableSpeed air-to-air coil autosizes cooling capacity using CapFT curve results based on RatedInletWaterTemp = 84.2
Issue overview
--------------
line 3357 of VariableSpeedCoil:
TotCapTempModFac = CurveValue( VarSpeedCoil( DXCoilNum ).MSCCapFTemp( VarSpeedCoil( DXCoilNum ).NormSpedLevel ), MixWetBulb, RatedInletWaterTemp );

Found during development of unit test = UnitarySystem_MultispeedPerformance

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
-------------------------------

- Fix #6204 
- Coil:Cooling:DX:VariableSpeed air-to-air coil should autosize cooling capacity using CapFT curve results based on `OutTemp` and not `RatedInletWaterTemp`

I modified an existing unit test `ZoneUnitarySysTest.UnitarySystemModel_MultispeedPerformance` that had a workaround in place which was to make the `DataSizing::DesDayWeath(1).Temp(1) = 29.4;` the same as the `RatedInletWaterTemp` (which @rraustad did when he found the issue and reported it)

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 1]
```diff
@@ -2431,16 +2428,18 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
     EXPECT_EQ(thisSys->m_MaxCoolAirVolFlow, 1.5);
     EXPECT_EQ(thisSys->m_MaxHeatAirVolFlow, 1.5);
 
-    // TotCapTempModFac is evaluated at the OutTemp which is 29.4°C (which is the same at the RatedInletWaterTemperature - work around for #6204)
-    Real64 RatedSourceTempCool = 29.4;
+    // TotCapTempModFac is evaluated at the OutTemp which is 35°C
+    // In the Fixture's SetUp: `DataSizing::DesDayWeath(1).Temp(1) = 35.0`
+    Real64 RatedSourceTempCool = DataSizing::DesDayWeath(1).Temp(1);
+    EXPECT_EQ(RatedSourceTempCool, 35.0);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc:2792]: Once I remove the workaround, this test now fails.

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 2]
```diff
@@ -4081,8 +4081,13 @@ namespace VariableSpeedCoils {
...
                     MixWetBulb = PsyTwbFnTdbWPb(MixTemp, MixHumRat, OutBaroPress, RoutineName);
                     // need to use OutTemp for air-cooled and RatedInletWaterTemp for water-cooled
+                    if (VarSpeedCoil(DXCoilNum).CondenserInletNodeNum != 0) {
+                        RatedSourceTempCool = RatedInletWaterTemp;
+                    } else {
+                        RatedSourceTempCool = OutTemp;
+                    }
                     TotCapTempModFac =
-                        CurveValue(VarSpeedCoil(DXCoilNum).MSCCapFTemp(VarSpeedCoil(DXCoilNum).NormSpedLevel), MixWetBulb, RatedInletWaterTemp);
+                        CurveValue(VarSpeedCoil(DXCoilNum).MSCCapFTemp(VarSpeedCoil(DXCoilNum).NormSpedLevel), MixWetBulb, RatedSourceTempCool);
```
Comment (jmarrec) [inline src/EnergyPlus/VariableSpeedCoils.cc:4090]: Fix 1 out of 2

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 3]
```diff
@@ -4153,8 +4158,13 @@ namespace VariableSpeedCoils {
...
```
Comment (jmarrec) [inline src/EnergyPlus/VariableSpeedCoils.cc:4167]: Fix 2 out of 2

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 4]
```diff
@@ -2310,9 +2311,6 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
     FirstHVACIteration = false;
     DataGlobals::BeginEnvrnFlag = false;
 
-    // overwrite outdoor weather temp to variable speed coil rated water temp until this gets fixed
-    DataSizing::DesDayWeath(1).Temp(1) = 29.4;
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc:2672]: Old workaround.

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 5]
```diff
@@ -2429,36 +2427,51 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
...
+                                            RatedSourceTempCool);
+
+    EXPECT_NEAR(TotCapTempModFac, 0.930018048445091, 0.001);
+    Real64 RatedCapCoolTotalDes = CoolCoolCapAtPeak / TotCapTempModFac;
+    EXPECT_NEAR(RatedCapCoolTotalDes, 34897.0396944, 0.001);
+
+    EXPECT_NEAR(thisSys->m_DesignCoolingCapacity, RatedCapCoolTotalDes, 0.001);
     EXPECT_EQ(thisSys->m_DesignCoolingCapacity, VariableSpeedCoils::VarSpeedCoil(1).RatedCapCoolTotal);
-    EXPECT_NEAR(thisSys->m_DesignHeatingCapacity, 32752.541, 0.001);
+    EXPECT_NEAR(thisSys->m_DesignHeatingCapacity, RatedCapCoolTotalDes, 0.001);
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc:2804]: Updated values due to using the right OutTemp for the curve evaluation

File: src/EnergyPlus/VariableSpeedCoils.cc (hunk 2 above)
Comment (mitchute) [reply src/EnergyPlus/VariableSpeedCoils.cc:4090]: 👍

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 6]
```diff
@@ -2787,36 +2785,51 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
...
     EXPECT_EQ(VariableSpeedCoils::VarSpeedCoil(1).RatedAirVolFlowRate,
               VariableSpeedCoils::VarSpeedCoil(1).RatedCapCoolTotal * VariableSpeedCoils::VarSpeedCoil(1).MSRatedAirVolFlowPerRatedTotCap(10));
-    EXPECT_NEAR(VariableSpeedCoils::VarSpeedCoil(1).RatedAirVolFlowRate, 1.82964, 0.00001); // different than unitary system air volume flow rate
+    Real64 fullFlow = 1.949442;
+    EXPECT_NEAR(VariableSpeedCoils::VarSpeedCoil(1).RatedAirVolFlowRate, fullFlow, 0.00001); // different than unitary system air volume flow rate
     EXPECT_NEAR(VariableSpeedCoils::VarSpeedCoil(2).RatedAirVolFlowRate, 1.70, 0.01);       // VS DX heating coil was not autosized
 
     // checks on autosized cooling air flow rates
-    EXPECT_NEAR(thisSys->m_CoolVolumeFlowRate[1], 0.182964, 0.000001);
+    EXPECT_NEAR(thisSys->m_CoolVolumeFlowRate[1], fullFlow * 0.1, 0.000001);
```
Comment (mitchute) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc:2814]: This is nicer.

File: src/EnergyPlus/VariableSpeedCoils.cc (hunk 3 above)
Comment (mitchute) [reply src/EnergyPlus/VariableSpeedCoils.cc:4167]: 👍

**Review by mitchute — APPROVED** [review]
Looks good. This should be able to merge shortly.

**Comment by mitchute** [comment]
CI is generally happy aside from the results diffs, and they all appear to be justified due to the change to the sizing condition. Merging.

**Pull request merged**